search/django/weedid/views.py

```diff
@@ -32,22 +32,30 @@ def upload(request):
     if request.method == "POST":
         user = request.user
         if user and user.is_authenticated:
-            images = []
-            file_weedcoco = request.FILES["weedcoco"]
-            weedcoco_json = json.load(file_weedcoco)
-            validate(weedcoco_json)
-            for image_reference in weedcoco_json["images"]:
-                images.append(image_reference["file_name"].split("/")[-1])
-            upload_dir, upload_id = setup_upload_dir(
-                os.path.join(UPLOAD_DIR, str(user.id))
-            )
-            weedcoco_path = store_tmp_weedcoco(file_weedcoco, upload_dir)
-            create_upload_entity(weedcoco_path, upload_id, user.id)
-            return HttpResponse(json.dumps({"upload_id": upload_id, "images": images}))
+            try:
+                images = []
+                file_weedcoco = request.FILES["weedcoco"]
+                weedcoco_json = json.load(file_weedcoco)
+                validate(weedcoco_json)
```

Comment:
FWIW this might be slow, so we might want a less synchronous way of performing it...?

Ideally we'd call `validate` with the set of images so that it can check that the image sizes match the COCO.